diff --git a/node_modules/@react-native/gradle-plugin/.gradle/buildOutputCleanup/buildOutputCleanup.lock b/node_modules/@react-native/gradle-plugin/.gradle/buildOutputCleanup/buildOutputCleanup.lock
new file mode 100644
index 0000000..5a051aa
Binary files /dev/null and b/node_modules/@react-native/gradle-plugin/.gradle/buildOutputCleanup/buildOutputCleanup.lock differ
diff --git a/node_modules/@react-native/gradle-plugin/.gradle/buildOutputCleanup/cache.properties b/node_modules/@react-native/gradle-plugin/.gradle/buildOutputCleanup/cache.properties
new file mode 100644
index 0000000..a150835
--- /dev/null
+++ b/node_modules/@react-native/gradle-plugin/.gradle/buildOutputCleanup/cache.properties
@@ -0,0 +1,2 @@
+#Tue Jun 16 08:17:49 UTC 2026
+gradle.version=9.3.1
diff --git a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/build.gradle.kts b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/build.gradle.kts
index 85dadeb..0fbc63b 100644
--- a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/build.gradle.kts
+++ b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/build.gradle.kts
@@ -60,7 +60,7 @@ dependencies {
 // nice message showing that JDK 11 (or 17) is required first
 java { targetCompatibility = JavaVersion.VERSION_11 }

-kotlin { jvmToolchain(17) }
+kotlin {  }

 tasks.withType<KotlinCompile>().configureEach {
   compilerOptions {
diff --git a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/JdkConfiguratorUtils.kt b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/JdkConfiguratorUtils.kt
index 151b53e..3f13b79 100644
--- a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/JdkConfiguratorUtils.kt
+++ b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/JdkConfiguratorUtils.kt
@@ -38,8 +38,8 @@ internal object JdkConfiguratorUtils {
             project.extensions
                 .getByType(ApplicationAndroidComponentsExtension::class.java)
                 .finalizeDsl { ext ->
-                  ext.compileOptions.sourceCompatibility = JavaVersion.VERSION_17
-                  ext.compileOptions.targetCompatibility = JavaVersion.VERSION_17
+                  ext.compileOptions.sourceCompatibility = JavaVersion.VERSION_21
+                  ext.compileOptions.targetCompatibility = JavaVersion.VERSION_21
                 }
           }
       val libraryAction =
@@ -47,17 +47,17 @@ internal object JdkConfiguratorUtils {
             project.extensions
                 .getByType(LibraryAndroidComponentsExtension::class.java)
                 .finalizeDsl { ext ->
-                  ext.compileOptions.sourceCompatibility = JavaVersion.VERSION_17
-                  ext.compileOptions.targetCompatibility = JavaVersion.VERSION_17
+                  ext.compileOptions.sourceCompatibility = JavaVersion.VERSION_21
+                  ext.compileOptions.targetCompatibility = JavaVersion.VERSION_21
                 }
           }
       project.pluginManager.withPlugin("com.android.application", applicationAction)
       project.pluginManager.withPlugin("com.android.library", libraryAction)
       project.pluginManager.withPlugin("org.jetbrains.kotlin.android") {
-        project.kotlinExtension.jvmToolchain(17)
+        project.kotlinExtension.jvmToolchain(21)
       }
       project.pluginManager.withPlugin("org.jetbrains.kotlin.jvm") {
-        project.kotlinExtension.jvmToolchain(17)
+        project.kotlinExtension.jvmToolchain(21)
       }
     }
   }
diff --git a/node_modules/@react-native/gradle-plugin/settings-plugin/build.gradle.kts b/node_modules/@react-native/gradle-plugin/settings-plugin/build.gradle.kts
index 2fbeafd..48ac01e 100644
--- a/node_modules/@react-native/gradle-plugin/settings-plugin/build.gradle.kts
+++ b/node_modules/@react-native/gradle-plugin/settings-plugin/build.gradle.kts
@@ -50,7 +50,7 @@ dependencies {
 // nice message showing that JDK 11 (or 17) is required first
 java { targetCompatibility = JavaVersion.VERSION_11 }

-kotlin { jvmToolchain(17) }
+kotlin {  }

 tasks.withType<KotlinCompile>().configureEach {
   compilerOptions {
diff --git a/node_modules/@react-native/gradle-plugin/settings.gradle.kts b/node_modules/@react-native/gradle-plugin/settings.gradle.kts
index b3c46a3..7483484 100644
--- a/node_modules/@react-native/gradle-plugin/settings.gradle.kts
+++ b/node_modules/@react-native/gradle-plugin/settings.gradle.kts
@@ -13,8 +13,6 @@ pluginManagement {
   }
 }

-plugins { id("org.gradle.toolchains.foojay-resolver-convention").version("0.5.0") }
-
 include(
     ":react-native-gradle-plugin",
     ":settings-plugin",
diff --git a/node_modules/@react-native/gradle-plugin/shared-testutil/build.gradle.kts b/node_modules/@react-native/gradle-plugin/shared-testutil/build.gradle.kts
index 26173fb..7e10415 100644
--- a/node_modules/@react-native/gradle-plugin/shared-testutil/build.gradle.kts
+++ b/node_modules/@react-native/gradle-plugin/shared-testutil/build.gradle.kts
@@ -23,7 +23,7 @@ dependencies { implementation(libs.junit) }

 java { targetCompatibility = JavaVersion.VERSION_11 }

-kotlin { jvmToolchain(17) }
+kotlin {  }

 tasks.withType<KotlinCompile>().configureEach {
   compilerOptions {
diff --git a/node_modules/@react-native/gradle-plugin/shared/build.gradle.kts b/node_modules/@react-native/gradle-plugin/shared/build.gradle.kts
index 3159848..e443501 100644
--- a/node_modules/@react-native/gradle-plugin/shared/build.gradle.kts
+++ b/node_modules/@react-native/gradle-plugin/shared/build.gradle.kts
@@ -29,7 +29,7 @@ dependencies {

 java { targetCompatibility = JavaVersion.VERSION_11 }

-kotlin { jvmToolchain(17) }
+kotlin {  }

 tasks.withType<KotlinCompile>().configureEach {
   compilerOptions {
